core/src/test/java/org/elasticsearch/common/io/stream/BytesStreamsTests.java

```diff
@@ -812,4 +812,34 @@ public void testVLong() throws IOException {
         StreamInput in = new BytesArray(Base64.getDecoder().decode("////////////AQAAAAAAAA==")).streamInput();
         assertEquals(-1, in.readVLong());
     }
+
+    public enum TestEnum {
+        ONE,
+        TWO,
+        THREE
+    }
+
+    public void testEnum() throws IOException {
+        TestEnum value = randomFrom(TestEnum.values());
```

Comment:
Thank you @jasontedor,
I'll have a closer look at how the tests are run.
I assume it runs each test method several times, otherwise I would still suggest to restructure `#testInvalidEnum` so it always asserts the invalid enum case.